### ISSUE TITLE
Compatible with turborepo or nx?

### ISSUE THREAD

**Issue by snorreks** [rxdi/firelink#43]
Wondering if firelink can be used with turborepo or nx. Love to see an example if anyone has.

**Comment by Stradivario** [comment]
Hello there and thank you for the ticket!

I think nx using `affected` is doing the same thing but it saves the packages to dist folder and i think it maps them to node_modules. Never try with nx since mostly i am using `nx` with Angular and NestJS deployed in a docker container. I need to try it out and how it will behave.

But what do you want to achieve in this situation is crucial if we are deploying a Lambda service for example it should be pretty much the same flow of using `firelink` there shouldn't be any changes.

Turborepo is something that i found out recently and i need to check it out haven't tested it  for now and what are the differences regarding regular monorepo flow.

Cheers and good discussion! :)

**Comment by snorreks** [comment]
Thank you for the quick response! I am planning to migrate to monorepo, either turborepo or nx, but I can't start unless I know that it is possible to use shared libraries with firebase functions. If you find out anything, please let me know.

**Comment by WesleyYue** [comment]
We are using Firelink with turborepo perfectly fine. There's nothing special about turborepo. It's just Yarn/PNPM workspaces underneath.

@Stradivario Thank you for putting this package together. I can't believe the firebase team just let this issue rot for over 5 years.

**Comment by Stradivario** [comment]
> We are using Firelink with turborepo perfectly fine. There's nothing special about turborepo. It's just Yarn/PNPM workspaces underneath.
> 
> @Stradivario Thank you for putting this package together. I can't believe the firebase team just let this issue rot for over 5 years.

You don't really know how thankful i am to read your comment!

I really appreciate it and living to help others :))

I think we can create far more nice features regarding firelink but it becomes useless if firebase team introduces fix for monorepo deployment which doesn't happen like you say from about 5 years...

I used firebase functions for about 2 years back then and after some really big payments that i needed to introduce i decided to create my own framework for building lambdas.
I have tried many different lambda architectures and non of them was suitable for me so i decided that it is best to create my own cluster on Kubernetes where i can deploy many lambdas as my wish without too much of a price.

So again thank you for your comment it really made my day! 

Cheers!

@WesleyYue
